lib/db/instance.go

```diff
@@ -506,6 +506,9 @@ func (db *instance) getIndexID(device, folder []byte) protocol.IndexID {
 func (db *instance) setIndexID(device, folder []byte, id protocol.IndexID) {
 	bs, _ := id.Marshal() // marshalling can't fail
 	if err := db.Put(db.keyer.GenerateIndexIDKey(nil, device, folder), bs, nil); err != nil {
+		if err == leveldb.ErrClosed {
+			return
+		}
```

Comment:
Why is it needed only here?

Reply:
Because in other places our error checking amounts to something like
```
	if err != nil {
		l.Debugln("surprise error:", err)
		return nil
	}
```
which I consider an adequate reaction to being closed and doesn't need more specific checks (and it doesn't result in a panic).

Reply:
There's also stuff like db writes where we just straight up panic on any error

Reply:
Now all panics in lib/db are either removed or have a check for close errors (or are not related to goleveldb in the first place).